### PR TITLE
ui: show personality updates

### PR DESCRIPTION
**Pull request by royjr**
untested pr for #33404

### CODE REVIEW

File: selfdrive/controls/lib/events.py [hunk 1]
```diff
@@ -325,6 +325,14 @@ def joystick_alert(CP: car.CarParams, CS: car.CarState, sm: messaging.SubMaster,
   vals = f"Gas: {round(gb * 100.)}%, Steer: {round(steer * 100.)}%"
   return NormalPermanentAlert("Joystick Mode", vals)
 
+def personality_changed_alert(CP: car.CarParams, CS: car.CarState, sm: messaging.SubMaster, metric: bool, soft_disable_time: int) -> Alert:
+  personality = sm['controlsState'].personality
+  return Alert(
+    "Personality:",
+    f"{personality}",
```
Comment (adeebshihadeh) [inline selfdrive/controls/lib/events.py:332]: should be a one liner

File: selfdrive/controls/lib/events.py [hunk 2]
```diff
@@ -325,6 +325,14 @@ def joystick_alert(CP: car.CarParams, CS: car.CarState, sm: messaging.SubMaster,
...
+    AlertStatus.userPrompt, AlertSize.mid,
+    Priority.LOW, VisualAlert.steerRequired, AudibleAlert.promptRepeat, 2.)
```
Comment (adeebshihadeh) [inline selfdrive/controls/lib/events.py:334]: no visual or audible alert. also maybe 2s is too long?

**Review by adeebshihadeh — CHANGES_REQUESTED** [review]

File: selfdrive/controls/lib/events.py [hunk 3]
```diff
@@ -325,6 +325,22 @@ def joystick_alert(CP: car.CarParams, CS: car.CarState, sm: messaging.SubMaster,
   vals = f"Gas: {round(gb * 100.)}%, Steer: {round(steer * 100.)}%"
   return NormalPermanentAlert("Joystick Mode", vals)
 
+def personality_changed_alert(CP: car.CarParams, CS: car.CarState, sm: messaging.SubMaster, metric: bool, soft_disable_time: int) -> Alert:
+  personality = log.LongitudinalPersonality
```
Comment (sshane) [inline selfdrive/controls/lib/events.py:329]: make global constant

File: selfdrive/controls/lib/events.py [hunk 4]
```diff
@@ -325,6 +325,22 @@ def joystick_alert(CP: car.CarParams, CS: car.CarState, sm: messaging.SubMaster,
...
+  personality_text = "unknown"
+
+  if personality == log.LongitudinalPersonality.relaxed:
+    personality_text = "relaxed"
+  elif personality == log.LongitudinalPersonality.standard:
+    personality_text = "standard"
+  elif personality == log.LongitudinalPersonality.aggressive:
+    personality_text = "aggressive"
```
Comment (sshane) [inline selfdrive/controls/lib/events.py:337]: just do str() on the field

**Comment by royjr** [comment]
it works but probably need to add `control_state: log.ControlsState` to the rest of the alerts like I did to `startup_master_alert`

File: selfdrive/controls/controlsd.py [hunk 5]
```diff
@@ -740,45 +742,45 @@ def publish_logs(self, CS, start_time, CC, lac_log):
     # controlsState
     dat = messaging.new_message('controlsState')
     dat.valid = CS.canValid
-    controlsState = dat.controlsState
+    self.controlsState = dat.controlsState
```
Comment (adeebshihadeh) [inline selfdrive/controls/controlsd.py:745]: don't modify all these lines

File: selfdrive/controls/lib/events.py [hunk 6]
```diff
@@ -325,6 +325,22 @@ def joystick_alert(CP: car.CarParams, CS: car.CarState, sm: messaging.SubMaster,
   vals = f"Gas: {round(gb * 100.)}%, Steer: {round(steer * 100.)}%"
   return NormalPermanentAlert("Joystick Mode", vals)
 
+def personality_changed_alert(CP: car.CarParams, CS: car.CarState, sm: messaging.SubMaster, metric: bool, soft_disable_time: int, control_state: log.ControlsState) -> Alert:
+  personality = control_state.personality
+  personality_text = "unknown"
+
+  if personality == log.LongitudinalPersonality.relaxed:
+    personality_text = "Standard"
```
Comment (adeebshihadeh) [inline selfdrive/controls/lib/events.py:333]: str(personality).upper()

**Comment by adeebshihadeh** [comment]
Thanks @royjr.

**Pull request merged**